src/librustc_typeck/collect.rs

```diff
@@ -1182,14 +1182,28 @@ fn generics_of(tcx: TyCtxt<'_>, def_id: DefId) -> &ty::Generics {
         }
         // FIXME(#43408) enable this always when we get lazy normalization.
         Node::AnonConst(_) => {
+            let parent_id = tcx.hir().get_parent_item(hir_id);
```

Comment:
Nit: bind `let hir = tcx.hir();` earlier in the function to reduce repetition.

Reply:
What's funny is that I copy-pasted these two lines from #70452 - but also I'm not fond of binding `tcx.hir()` due to the ideal future in which we request the `tcx.hir(...)` for a given `LocalDefId` and then use that, binding it wouldn't help with that AFAICT.

Reply:
Well it's just a nit, but are there plans for that future though?